trieste/acquisition/rule.py

```diff
@@ -290,7 +293,75 @@ def acquire(
             )
 
         acquisition_function = self._builder.prepare_acquisition_function(datasets, models)
+
         point = _optimizer.optimize(acquisition_space, acquisition_function)
         state_ = TrustRegion.State(acquisition_space, eps, y_min, is_global)
 
         return point, state_
+
+
+class BatchAcquisitionRule(AcquisitionRule[None, SearchSpace]):
+    """ Implements a acquisition rule for a batch of query points """
+
+    def __init__(self, num_query_points: int,
+                 builder: Optional[AcquisitionFunctionBuilder] = None):
+        """
+        :param num_query_points: The number of points to acquire.
+        :param builder: The acquisition function builder to use.
+            :class:`EfficientGlobalOptimization` will attempt to **maximise** the corresponding
+            acquisition function. Defaults to :class:`~trieste.acquisition.ExpectedImprovement`
+            with tag `OBJECTIVE`.
+        """
+
+        if not num_query_points > 0:
+            raise ValueError(
+                f"Number of query points must be greater than 0, got {num_query_points}"
+            )
+
+        self._num_query_points = num_query_points
+
+        if builder is None:
+            num_samples = 100
+            builder = MonteCarloExpectedImprovement(eps_shape=[num_samples, num_query_points, 1]).using(OBJECTIVE)  # [S, B, L]
+
+        self._builder = builder
+
+    def expand_search_space(self, search_space: SearchSpace):
+        assert isinstance(search_space, Box)
+
+        lower_bound, upper_bound = search_space.lower, search_space.upper
+        expanded_lower_bound = tf.tile(lower_bound, [self._num_query_points])
+        expanded_upper_bound = tf.tile(upper_bound, [self._num_query_points])
+        return Box(expanded_lower_bound, expanded_upper_bound)
+
+    def vectorize_batch_acquisition(self, acquisition_function, dim):
+        return lambda at: acquisition_function(tf.reshape(at, [-1, self._num_query_points, dim]))
```

Comment:
do users need these methods (can you give an example)? if not, can we make them private?

Reply:
I think that it can be useful for plotting, post-optim analysis, etc.

Reply:
happy to make them private if you think it's important (but why is it important?)

Reply:
public methods are features, so lots of public methods indicate scope creep

Reply:
to think about it another way: there are several ways to make this functionality available, so we need to think about which is best. That may not take very long, but it is extra work that's not needed to complete this work. Similarly, public functionality needs tests - we'll need tests for every public method. That tends to take time, so we need to ask if this is the best use of our time